### DISCUSSION TITLE
Mesh colored on "top" and "bottom" of the surface?

### DISCUSSION THREAD

**Discussion by ThomasDaheim** [FXyz/FXyz#168] [General]
And (hopefully) the last question in my list:

I would like to have the color of the mesh applied to the "top" and "bottom" of the surface so that it looks the same from all directions. Sounds really simple, but I didn't manage to achieve that. Is there any way to do that?

**Comment by Birdasaur** [comment]
Hey @ThomasDaheim Do you still need this resolved?
What mesh are you trying to color again? A SurfacePlotMesh?

**Comment by ThomasDaheim** [comment]
Hey @Birdasaur !

I'm using SurfacePlotMesh I would still be interested in that feature. But its not a "must have". It only looks strange if you view the plot from "below":

![Clipboard01](https://user-images.githubusercontent.com/14020617/211365902-ee591d08-8e6d-4fc0-9089-fed30ca5adef.jpg)
![Clipboard02](https://user-images.githubusercontent.com/14020617/211365928-cadd4783-5d9c-4354-8522-73ba3ca12aea.jpg)

**Comment by Birdasaur** [comment]
Technically you should be able to see the image from the bottom if you do not set the cull face but I think part of your problem is that your Y axis rises are so dense that its simply mudding up the color seen below? Or is that not the case? (I haven't dug into the lower level code for SurfacePlotMesh in a while.)
Maybe we need to improve it!

Ok this might be a hack but I'm just throwing out quick ideas...
Could you make a duplicate surface mesh in the same position but CullFace is Front so that the textures only show out the back?
Or a bigger hack you could make an inverted SurfaceMesh so that all your vertices reordered such that the "Front" is underneath. The geometry will LOOK the same but it would be painted inversely?

**Comment by ThomasDaheim** [comment]
Thanks for the ideas! I'm a bit busy with other projects (and live in general) at the moment. Might take some time before I manage to try that.

As for " I think part of your problem is that your Y axis rises are so dense that its simply mudding up the color seen below": That should be fairly easy to verify. Isn't there a SurfacePlotMesh out there somewhere to check that?

**Comment by Birdasaur** [comment]
I'm going to convert this to a discussion since we haven't identified a fix or upgrade to the code base yet but if we do feel free to make a new issue or add something to an existing one.
